validate/validate_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func TestGetMajorMinor(t *testing.T) {
+	cases := []struct {
+		version string
+		major   int
+		minor   int
+		err     error
+	}{
+		{"0.1beta", 0, 1, nil},
+		{"0.1.2", 0, 1, nil},
+		{"-1.-1beta", -1, -1, nil},
+		{"0.1", -1, -1, fmt.Errorf("have error")},
+		{"0", -1, -1, fmt.Errorf("have error")},
+		{"beta", -1, -1, fmt.Errorf("have error")},
+	}
+
+	for _, c := range cases {
+		ma, mi, e := getMajorMinor(c.version)
+		if (e != nil) != (c.err != nil) {
+			t.Errorf("Unexpected err, should %v, but got %v", c.err, e)
```

Comment:
I like to put the test case number in all error messages, e.g. `"[%d] ....", i`, where `i` is the first value returned by `range`